### PR TITLE
win,fs: make rename retries opt-in

### PR DESCRIPTION
**Pull request by refack**
Fixes: https://github.com/libuv/libuv/issues/2098

### CODE REVIEW

**Comment by refack** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1119/

**Comment by bnoordhuis** [comment]
I think it's practically a given that either:

1. No one will use the flag, or
2. People will request that we make the number of retries and the back-off period configurable.

If (1), it's dead code. If (2), it'd be hard to retrofit without breaking API/ABI.

FWIW, I agree with the sentiment from #2098 that retry behavior doesn't really have a place in libuv. I move for reverting it.

**Comment by refack** [comment]
> I agree with the sentiment from #2098 that retry behavior doesn't really have a place in libuv.

I'm ambivalent as well...

On it's face arbitrary-retry-on-fail seems like semantics that are a departure from libuv's expected behavior (even node-core pushed back on these, deferring to user-land `graceful-fs`). Also why just "rename", if we're venturing into this, for completeness I'd expect it implemented in open & copy as well...

On the other hand, the Windows built-in anti-virus and indexer are everywhere and on-by-default. They are so bothersome that almost every decent software product asks it's users to disables those as the first step of troubleshooting.

tl;dr This PR was made mainly to solicit feedback.

**Comment by bzoz** [comment]
If we hide this behind a flag Node will probably end up using that flag, so I guess it won't end up as dead code.

IMHO this should be made opt-out. Libuv should help with all those little platform differences. This "failing rename" is one of those things (and the "rename overwriting existing empty directory" is another). For most users, this makes libuv more reliable. For all others using new `UV_FS_RAW_RENAME` flag should work just fine.

As for jitter, incremental back off and not blocking a thread - all those can be implementend by extending current implementation.

**Comment by bzoz** [comment]
Retrying was reverted by https://github.com/libuv/libuv/pull/2122, closing this.
